overlord/assertstate/assertmgr.go

```diff
@@ -234,17 +236,30 @@ func (e *commitError) Error() string {
 	return fmt.Sprintf("cannot add some assertions to the system database:%s", strings.Join(l, "\n - "))
 }
 
+func clarifyUnsupportedFormat(err error, a asserts.Assertion) (clarifyd error) {
+	if _, ok := err.(*asserts.UnsupportedFormatError); ok {
+		return fmt.Errorf("%v assertion format is too new for this snapd, upgrade snapd/core snap", a.Ref())
```

Comment:
I think we should just return err here and tailor the underlying err to provide the needed information.